### PR TITLE
Gutenberg: remove dependency on calypso's jetpack logo component

### LINKED ISSUE #31193: Replace Jetpack logo component with inline SVG loader
This is part of removing any internal dependencies that are not explicit packages from Jetpack blocks.

### PR DESCRIPTION
**Pull request by roccotripaldi**
#### Changes proposed in this Pull Request

* Creates our own Jetpack logo component as we prepare to move these blocks out of the calypso repo and into the jetpack repo

#### Testing instructions

* Try out this [Jurassic.ninja link](https://jurassic.ninja/create/?gutenpack&shortlived&jetpack-beta&calypsobranch=fix/31193-jetpack-logo-dependency)
* "Setup Jetpack" on your testing site
* Go to the post editor.
* How do those Jetpack logos look?

screen shot 2019-03-04 at 2 46 19 pm

Fixes #31193

### CODE REVIEW

**Comment by lezama** [comment]
@Automattic/gutenpack is a Jetpack Logo Component a good case for something that should be shared via a Calypso package instead of being moved to Jetpack?

**Comment by simison** [comment]
>is a Jetpack Logo Component a good case for something that should be shared via a Calypso package instead of being moved to Jetpack?

I'd be inclined to not publish it as a package and just use the file from Jetpack.

I don't see icon SVG here; https://github.com/Automattic/jetpack/tree/ada24ea2b6e173f77ecb4f9c32393d72554013cc/images

Prolly good first to copy the file to `client/gutenberg/extensions/presets/jetpack` and use it from there, so that we cut the connection to Calypso?

File: client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import { Path, Polygon, SVG } from '@wordpress/components';
+import classNames from 'classnames';
+
+const JetpackLogo = ( { size = 32, className } ) => (
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx]: I don't think we need any props
```suggestion
const JetpackLogo = () => (
```

File: client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	<SVG
+		className={ classNames( 'jetpack-logo', className ) }
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx]: We could keep `jetpack-logo` className, but I don't think we need it…
```suggestion
```

File: client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+		width={ size }
+		height={ size }
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx]:
```suggestion
		height={ 24 }
```

File: client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import { Path, Polygon, SVG } from '@wordpress/components';
+import classNames from 'classnames';
+
+const JetpackLogo = ( { size = 32, className } ) => (
+	<SVG
+		className={ classNames( 'jetpack-logo', className ) }
+		width={ size }
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx]: Gutenberg will resize to 24 anyways, I believe.
```suggestion
		width={ 24 }
```

File: client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+		height={ size }
+		viewBox="0 0 32 32"
+	>
+		<Path
+			className="jetpack-logo__icon-circle"
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx]:
```suggestion
```

File: client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+			fill="#00be28"
+			d="M16,0C7.2,0,0,7.2,0,16s7.2,16,16,16s16-7.2,16-16S24.8,0,16,0z"
+		/>
+		<Polygon className="jetpack-logo__icon-triangle" fill="#fff" points="15,19 7,19 15,3 " />
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx]:
```suggestion
		<Polygon fill="#fff" points="15,19 7,19 15,3 " />
```

File: client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+		<Polygon className="jetpack-logo__icon-triangle" fill="#fff" points="17,29 17,13 25,13 " />
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx]:
```suggestion
		<Polygon fill="#fff" points="17,29 17,13 25,13 " />
```

File: client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import { Path, Polygon, SVG } from '@wordpress/components';
+import classNames from 'classnames';
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx]:
```suggestion
```

**Comment by lezama** [comment]
Applied all @sirreal suggestions.

I don't see any regressions in Calypso

screen shot 2019-03-05 at 08 51 49

File: client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+		width={ 24 }
+		height={ 24 }
+		viewBox="0 0 32 32"
+	>
+		<Path
+			fill="#00be28"
+			d="M16,0C7.2,0,0,7.2,0,16s7.2,16,16,16s16-7.2,16-16S24.8,0,16,0z"
+		/>
+		<Polygon fill="#fff" points="15,19 7,19 15,3 " />
+		<Polygon fill="#fff" points="17,29 17,13 25,13 " />
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/utils/jetpack-logo.jsx]: There's a tweaked version for 24px size:

https://github.com/Automattic/wp-calypso/blob/241292989efa898345417b4632f6402605497895/client/components/jetpack-logo/index.jsx#L38-L45

If that is truly the size that Gutenberg uses, we should consider swapping it. If I recally correctly, @shaunandrews may have details on whether/why this tweaked version was/continues to be necessary 🙇

**Comment by roccotripaldi** [comment]
Oops, something has gone wonky with your latests changes. Here is the pre-publish icon now:

screen shot 2019-03-05 at 9 46 39 am

**Comment by roccotripaldi** [comment]
Also, I believe our component needs to allow `size` and `className` to be overwritten by core.
I believe core adds classes and sizes to any icons given to it by `setCategories`

with your changes:

screen shot 2019-03-05 at 9 46 17 am

with `( { size, className } )` overwritabble:

screen shot 2019-03-05 at 9 45 15 am

**Comment by roccotripaldi** [comment]
Here is where Gutenberg overrides props on the category icon:
https://github.com/WordPress/gutenberg/blob/master/packages/components/src/panel/body.js#L71

**Comment by roccotripaldi** [comment]
@sirreal - we've moved the component from the `utils` to `editor-shared` and we want to keep the class names and for now, and keep the logo's props extendable. 

we can perhaps revisit this after @ockham 's css pr lands.

what do you think?

File: client/gutenberg/extensions/presets/jetpack/editor-shared/jetpack-logo.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import { Path, Polygon, SVG } from '@wordpress/components';
+import classNames from 'classnames';
+
+export default ( { size = 32, className } ) => (
```
Comment (sirreal) [inline client/gutenberg/extensions/presets/jetpack/editor-shared/jetpack-logo.js]: NIT: We might use `20` or `24` as a default size, I don't think Gutenberg uses 32 anywhere.

```suggestion
export default ( { size = 24, className } ) => (
```

**Pull request merged**